modules/core/shared/src/main/scala/lucuma/core/model/sequence/gmos/binning.scala

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.model.sequence.gmos
+
+import cats.Order
+import cats.syntax.all.*
+import coulomb.Quantity
+import eu.timepit.refined.types.numeric.PosDouble
+import eu.timepit.refined.types.numeric.PosInt
+import lucuma.core.enums.GmosNorthDetector
+import lucuma.core.enums.GmosNorthFpu
+import lucuma.core.enums.GmosNorthGrating
+import lucuma.core.enums.GmosSouthDetector
+import lucuma.core.enums.GmosSouthFpu
+import lucuma.core.enums.GmosSouthGrating
+import lucuma.core.enums.GmosXBinning
+import lucuma.core.enums.GmosYBinning
+import lucuma.core.enums.ImageQuality
+import lucuma.core.math.Angle
+import lucuma.core.math.Wavelength
+import lucuma.core.math.units.NanometersPerPixel
+import lucuma.core.model.SourceProfile
+import spire.math.Rational
+
+import java.math.RoundingMode.HALF_UP
+
+/**
+ * Optimal GMOS spectral and spatial binning calculations.
+ *
+ * See https://docs.google.com/document/d/1P8_pXLRVomUSvofyVkAniOyGROcAtiJ7EMYt9wWXB0o.
+ */
+object binning {
+
+  val DefaultSampling: PosDouble =
+    PosDouble.unsafeFrom(2.5)
+
+  /**
+   * Object angular size estimate based on source profile and image quality.
+   */
+  def objectSize(p: SourceProfile, iq: ImageQuality): Angle =
+    p match {
+      case SourceProfile.Point(_)          => iq.toAngle
+      case SourceProfile.Uniform(_)        => Angle.Angle180
+      case SourceProfile.Gaussian(fwhm, _) => fwhm
+    }
+
+  extension (λ: Angle) {
+    def arcsec: BigDecimal =
+      Angle.decimalArcseconds.get(λ)
+  }
+
+  /**
+   * Minimum of the slit and object size.
+   */
+  def effectiveWidth(
+    slitWidth:  Angle,
+    srcProfile: SourceProfile,
+    iq:         ImageQuality
+  ): Angle = {
+    given Order[Angle] = Angle.AngleOrder
+    slitWidth min objectSize(srcProfile, iq)
```

Comment:
Ah I guess the 180 is intersected with slitWidth, right?

Reply:
For the spectral binning calculation yes.  In this case, the slit width sets the effective size.